### ISSUE TITLE
Wielditem coloring broken with shaders enabled

### ISSUE THREAD

**Issue by sfan5** [minetest/minetest#9941]
##### Minetest version
```
Minetest 5.3.0-dev-083b285f4
```

##### Summary

![screenshot_20200526_174703](https://user-images.githubusercontent.com/1042418/82928883-1c25e880-9f83-11ea-9aaf-ac870051ab05.png)

* Item coloring does not work for the wielditem (seen right)
* Firelike nodes are black when wielded (seen left)

both of these issues go away when reverting #9537
<b>this is a regression from 5.2.0</b>

##### Steps to reproduce
Take "Firelike Drawtype Test Node" and "Raillike Drawtype Test Node: rail 3" from devtest's chest and wield them.

**Comment by Wuzzy2** [comment]
@dcbrwn

**Comment by dcbrwn** [comment]
I'll take a look

**Comment by dcbrwn** [comment]
Apparently fire-like mesh didn't have normals. I've added a fallback for that but it might cause trouble:

```glsl
	// This is intentional comparison with zero without any margin.
	// If normal is not equal to zero exactly, then we assume it's a valid, just not normalized vector
	vIDiff = length(gl_Normal) == 0.0
		? 1.0
		: directional_ambient(normalize(gl_Normal));
```

As for coloring of rail-like type, I've fixed this in fragment shader, but there is still some issue with the mesh and I'm not sure what caused it. Looks like rail-like is rendered as a cube, with texture on the bottom. Most likely it's not related to #9537, but I'll double check to be sure.
